enhancements/upgrade-bundle.md

```diff
@@ -0,0 +1,138 @@
+* `--index-image=<>` is a required flag, which contains the operator we’re upgrading from
+* `--starting-csv=<>` is a required flag, which specifies the first version of the CSV that OLM will install as part of the subscription for a given package. 
+Here's an [sample CSV](https://github.com/operator-framework/operator-sdk-samples/blob/master/go/memcached-operator/deploy/olm-catalog/memcached-operator/manifests/memcached-operator.clusterserviceversion.yaml) in which the value of startingCSV is the name i.e. `memcached-operator.v0.17.0`
+* `[--channel=]` is an optional flag that specifies the channel to upgrade from within a subscription (default channel used if subscription does not contain a channel)
+* `[--operator-namespace=]` is an optional flag that specifies the namespace to install the operator. It will default to the `KUBECONFIG` context. (see run bundle)
+* `[--install-mode=]` is an optional flag that specifies the
+  [OLM install mode](https://red.ht/2YK0ksD). (see run bundle)
+  * We support 3 modes:
+    * AllNamespaces
+    * OwnNamespace
+    * SingleNamespace=ns1
+      * NOTE: if ns1 is the same as the operator's namespace, a warning will be
+        printed
+  * When `--install-mode` is not supplied, the default precedence order is:
+    * `OperatorGroup`, if it already exists
+    * `AllNamespaces`, if the CSV supports it
+    * `OwnNamespace`, if the CSV supports it
+    * If the CSV does not support either of these, `--install-mode` flag
+      will be **required**
+
+#### Assumptions
+* OLM is already installed
+* No previous version of the operator is already running
+* An index image exists that has the from version of the operator bundle
+* The provided images (i.e. bundle and index) are accessible and pullable by kubelet 
+
+#### Example UX Scenario
+
+```
+operator-sdk run bundle-upgrade <bundle-image> --index-image quay.io/operator-framework/upstream-opm-builder --starting-csv csv.v1 --channel alpha
+```
+
+ 1. Create pod using `--index-image`, set container command (entrypoint) to:
+    ```
+    /bin/bash -c ‘ \
+      /bin/mkdir -p /database && \
+      /bin/opm registry add   -d /database/index.db -b {.BundleImage} && \
+      /bin/opm registry serve -d /database/index.db’
+    ```
+ 1. If pod fails, clean it up, and error out with message of unsupported index image. Also capture the pod's logs for debugging.
+ 1. Create GRPC CatalogSource, pointing to `<pod-name>.<namespace>:50051`
+ 1. Create OperatorGroup, only if it does not exist
+ 1. Create Subscription
+    * Use `--channel` as "alpha" and `--starting-csv` as "csv.v1", taken from the flag values.
+    * Configure approval to automatic upgrades
+ 1. Operator installed into the default namespace defined by KUBECONFIG
+1. Verify operator is upgraded (by looking at the CSV version, status, etc.)
+1. Log what happened and final state (e.g. operator is running in namespace X, watching namespace(s) Y.
+1. Log command to execute to cleanup operator.
```

Comment:
What happens if an error occurs in one of the above steps? Do any resources get cleaned up?

Discuss this scenario in an implementation details section.